src/ott/neural/models.py

```diff
@@ -40,29 +40,34 @@ class ICNN(neuraldual.BaseW2NeuralDual):
   :cite:`amos:17` with initialization schemes proposed by :cite:`bunne:22`.
 
   Args:
-    dim_data: data dimensionality.
-    dim_hidden: sequence specifying size of hidden dimensions. The
-      output dimension of the last layer is 1 by default.
-    init_std: value of standard deviation of weight initialization method.
-    init_fn: choice of initialization method for weight matrices (default:
-      :func:`jax.nn.initializers.normal`).
-    act_fn: choice of activation function used in network architecture
-      (needs to be convex, default: :obj:`jax.nn.relu`).
-    pos_weights: Enforce positive weights with a projection.
-      If ``False``, the positive weights should be enforced with clipping
-      or regularization in the loss.
-    gaussian_map_samples: Tuple of source and target points, used to initialize
-      the ICNN to mimic the linear Bures map that morphs the (Gaussian
-      approximation) of the input measure to that of the target measure. If
-      ``None``, the identity initialization is used, and ICNN mimics half the
-      squared Euclidean norm.
+  dim_data: data dimensionality.
+  dim_hidden: sequence specifying size of hidden dimensions. The
+  output dimension of the last layer is 1 by default.
+  rank: rank of the quadratic layers. If ``0``, quadratic layers are not used.
```

Comment:
it seems that regardless of `rank==0`, quadratic layers will be used no? their quadratic component might by 0, but the linear is not